ansible_roles/roles/existing_uploads_dir/tasks/main.yml

```diff
@@ -0,0 +1,32 @@
+---
+- name: Check to see if the directory exist.
+  stat:
+    path: "/{{ config_info.user_parent_home_dir }}/{{ config_info.test_user }}/{{ type }}"
+  register: dir_stat_result
+- name: handle existing directory
+  block:
+  - name: Check to see if the zathras use file  exist.
+    stat:
+      path: "/{{ config_info.user_parent_home_dir }}/{{ config_info.test_user }}/{{ type }}/zathras_{{ type }}"
+    register: stat_result
+  - name: move dir if required
+    command: "mv /{{ config_info.user_parent_home_dir }}/{{ config_info.test_user }}/{{ type }} /{{ config_info.user_parent_home_dir }}/{{ config_info.test_user }}/{{ type }}_back_zathras"
+    when: not stat_result.stat.exists
+  when: dir_stat_result.stat.isdir is defined and dir_stat_result.stat.isdir
+
+- name: Make the uploads directory
+  file:
+    path: "{{ dyn_data.kit_upload_directory }}/{{ type }}"
+    state: directory
+    mode: 0777
+- name: touch the zathras_upload_file
+  file:
+    path: "{{ dyn_data.kit_upload_directory }}/{{ type }}/zathras_{{ type }}"
+    state: touch
+    mode: 0777
+
+- name: link the extras in
+  shell:
+    cmd: "ln -s {{ dyn_data.kit_upload_directory }}/{{ type }} /{{ config_info.user_parent_home_dir }}/{{ config_info.test_user }}/{{ type }}"
+    warn: false
+  ignore_errors: yes
```

Comment:
Consider using the [`file` module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html#parameter-state) to set a link.

That way we can get accurate changed states and when a fatal error occurs, we don't ignore it